plugins/woocommerce/includes/class-wc-cart.php

```diff
@@ -1959,8 +1977,15 @@ public function get_coupon_discount_amount( $code, $ex_tax = true ) {
 	 * @return float discount amount
 	 */
 	public function get_coupon_discount_tax_amount( $code ) {
-		$totals = $this->get_coupon_discount_tax_totals();
-		return wc_cart_round_discount( isset( $totals[ $code ] ) ? $totals[ $code ] : 0, wc_get_price_decimals() );
+		$totals     = $this->get_coupon_discount_tax_totals();
+		$tax_amount = 0;
+		foreach ( $totals as $key => $value ) {
+			if ( wc_is_same_coupon( $key, $code ) ) {
+				$tax_amount = $value;
+				break;
+			}
+		}
+		return wc_cart_round_discount( $tax_amount, wc_get_price_decimals(), PHP_ROUND_HALF_DOWN );
```

Comment:
I see there is a wc_get_tax_rounding_mode; should we use this here? Also, what do you think about fixing this in a separate PR to keep the scope of this PR related to the coupon codes cases

Reply:
do you want to share some testing steps to test the rounding changes?